### DISCUSSION TITLE
Segmentation fault in slint_windowrc_dispatch_event() from oleaut32.dll when using Slider

### DISCUSSION THREAD

**Discussion by mitchgrout** [slint-ui/slint#4680] [Q&A]
I have the following `.slint` file, along with a standard minimal C++ source file to create and run the UI:

```js
import { Slider } from "std-widgets.slint";
export component MainWindow inherits Window {
  Slider { }
}
```

I'm linking this against a `slint_cpp.dll` **built from source** of v1.4.1. When running my executable, it will immediately terminate with a non-zero code. Running it under gdb provides the following error:

```
warning: mincore\com\oleaut32\dispatch\ups.cpp(2122)\oleaut32.dll!00007FFC4B51A726: (caller: 00007FFC4B519B39) ReturnHr(1) tid(36a4) 8002801D Library not registered.

Thread 1 received signal SIGSEGV, Segmentation fault.
0x00007ffc169f285e in slint_windowrc_dispatch_event () from ...\slint_cpp.dll
```

Removing the slider allows the window to show and run correctly.

I do not have Qt installed to my system if this matters (compiled the .dll myself since the pre-built Windows DLL fails to start due to Qt missing).

**Comment by mitchgrout** [comment]
So, unlikely that anyone else will have this problem, but making a (somewhat embarrassing) answer for posterity.

I had installed slint to my system via the installer distributed via the releases page. However, I couldn't link against this due to Qt5 missing on my platform. So, I opted to build the library from source in order to remove the dependency on Qt5. 

In a stroke of genius, I then decided to copy my built copy of `slint_cpp.dll` into the install directory, overwriting the `.dll` stored therein. My thought being, my build tool was able to pick up on the library here, so updating the `.dll` will keep things working, while removing the dependency on Qt5. Windows really didn't like this, so ended up throwing up some segfaults. Removing the system installation of slint, and instead pointing directly to my built copy solved the issue.

**Reply by hunger** [reply]
Thanks for sharing, and no worry: We all did similar things:-)
